httpserver/service_node.cpp

```diff
@@ -93,6 +95,33 @@ static std::shared_ptr<request_t> make_push_request(std::string&& data) {
     return make_post_request("/v1/swarms/push", std::move(data));
 }
 
+template <class T>
+static bool verify_message(const T& msg, const char* error_message = nullptr) {
+    if (!util::validateTTL(msg.ttl)) {
+        if (error_message)
+            error_message = "Provided TTL is not valid";
```

Comment:
Not sure it does what you intended. You are modifying `error_message` here, not the original `error_msg`, correct? How about we just print a warning?

Reply:
OOps wanted to pass the pointer by reference.
The idea is to provide a useful error message to the user, so the exception will bubble up until it's caught, and the exception message used in the response body.

Reply:
(obviously can't be a reference since I want it to be optional)